terraform/main.tf

```diff
@@ -0,0 +1,64 @@
+terraform {
+  required_providers {
+    google = {
+      source  = "hashicorp/google"
+    }
+  }
+}
+
+locals {
+  cred = file("cred.json")
+  email = lookup(jsondecode(local.cred), "client_email")
+}
+
+provider "google" {
+  credentials = local.cred
+  project = var.project_id
+  region  = var.region
+  zone    = "us-central1-c"
+}
+
+# Define and deploy a workflow
+resource "google_workflows_workflow" "test_runner_workflow" {
+  name = "test-runner-workflow"
+  region = var.region
+  description = "Test runner workflow"
+  service_account = local.email
+  # Import workflow YAML file
+  source_contents = file("workflow.yaml")
+
+  depends_on = [google_project_service.workflows]
+}
+
+# Define and deploy a tasks queue
+resource "google_cloud_tasks_queue" "test_runner_tasks_queue" {
+  name = "test-runner-tasks-queue"
+  location = var.region
+
+  rate_limits {
+    max_concurrent_dispatches = 1000
+    max_dispatches_per_second = 500
+  }
+
+  retry_config {
+    max_attempts = 100
```

Comment:
Does this mean we'll attempt to run a test job 100 times, where each job could take hours? Would prefer 2, or maybe even none. 

We should also have a strategy for what to do if this fails, probably write to firestore with a failed test result?

Reply:
I set the max_attempts to 1, therefore, the other fields are not longer required.